napari/utils/context/_expressions.py

```diff
@@ -0,0 +1,597 @@
+    """This transformer converts an ast.expr into an :class:`Expr`.
+
+    Examples
+    --------
+    >>> tree = ast.parse('my_var > 11', mode='eval')
+    >>> tree = ExprTranformer().visit(tree)  # transformed
+    """
+
+    _SUPPORTED_NODES = frozenset(
+        k
+        for k, v in globals().items()
+        if isinstance(v, type) and issubclass(v, Expr)
+    )
+
+    # fmt: off
+    @overload
+    def visit(self, node: ast.expr) -> Expr: ...  # noqa
+    @overload
+    def visit(self, node: PassedType) -> PassedType: ...  # noqa
+    # fmt: on
+
+    def visit(self, node):
+        """Visit a node in the tree, transforming into Expr."""
+
+        if isinstance(
+            node,
+            (
+                ast.cmpop,
+                ast.operator,
+                ast.boolop,
+                ast.unaryop,
+                ast.expr_context,
+            ),
+        ):
+            # all operation types just get passed through
+            return node
+
+        # filter here for supported expression node types
+        type_ = type(node).__name__
+
+        if type_ not in ExprTranformer._SUPPORTED_NODES:
+            if sys.version_info < (3, 8) and type_ in self._PY37_CONSTS:
+                val = getattr(node, self._PY37_CONSTS[type_])
+                return Constant(val, lineno=1, col_offset=0)
+            raise SyntaxError(f"Type {type_!r} not supported")
+
+        # providing fake lineno and col_offset here rather than using
+        # ast.fill_missing_locations for typing purposes
+        kwargs: Dict[str, Any] = dict(lineno=1, col_offset=0)
+
+        for name, field in ast.iter_fields(node):
+            if isinstance(field, ast.expr):
+                kwargs[name] = self.visit(field)
+            elif isinstance(field, list):
+                kwargs[name] = [self.visit(item) for item in field]
+            else:
+                kwargs[name] = field
+        return globals()[type_](**kwargs)
+
+    # can drop after py3.7 support is gone
+    _PY37_CONSTS = {
+        'Num': 'n',
+        'Str': 's',
+        'Bytes': 's',
+        'NameConstant': 'value',
+    }
+
+
+class ExprSerializer(ast.NodeVisitor):
+    """Serializes an :class:`Expr` into a string.
+
+    Examples
+    --------
+    >>> expr = Expr.parse('a + b == c')
+    >>> print(expr)
+    'a + b == c'
+
+    or ... using this visitor directly:
+
+    >>> ser = ExprSerializer()
+    >>> ser.visit(expr)
+    >>> out = "".join(ser.result)
+    """
+
+    def __init__(self, node: Optional[Expr] = None) -> None:
+        self._result: List[str] = []
+
+        def write(*params):
+            for item in params:
+                if isinstance(item, ast.AST):
+                    self.visit(item)
+                elif item:
+                    self._result.append(item)
+
+        self.write = write
+
+        if node is not None:
+            self.visit(node)
+
+    def __str__(self):
+        return "".join(self._result)
```

Comment:
Hum, will that respect operator priority/associativity  ? 
Probably need to check with `(a and b) || (c or d)` vs `a and (b || c) or d`, or maybe that's ok because of the order in which you visit the AST.

I can do that later, but need to run in another call.

Reply:
yeah, i'm not respecting priority at the moment.   We could go as far as to say we don't support operator priority in our mini-language (but i also think that it tends to work out ok in most cases because of the AST tree ordering)